src/mito2/src/compaction.rs

```diff
@@ -300,20 +277,86 @@ impl CompactionScheduler {
             for waiter in waiters {
                 waiter.send(Ok(0));
             }
-            return None;
+            self.region_status.remove(&region_id);
+            return Ok(());
         };
 
-        let task = CompactionTaskImpl {
+        // If specified to run compaction remotely, we schedule the compaction job remotely.
+        // It will fall back to local compaction if there is no remote job scheduler.
+        if current_version.options.compaction.remote_compaction() {
+            if let Some(remote_job_scheduler) = &self.plugins.get::<RemoteJobSchedulerRef>() {
+                let remote_compaction_job = CompactionJob {
+                    compaction_region: compaction_region.clone(),
+                    picker_output: picker_output.clone(),
+                    start_time,
+                };
+
+                let result = remote_job_scheduler
+                    .schedule(
+                        RemoteJob::CompactionJob(remote_compaction_job),
+                        Arc::new(DefaultNotifier {
+                            request_sender: request_sender.clone(),
+                        }),
+                    )
+                    .await;
+
+                if let Ok(job_id) = result {
+                    info!(
+                        "Scheduled remote compaction job {} for region {}",
+                        job_id.as_u64(),
+                        region_id
+                    );
+
+                    // Return 0 to the client as we are going to schedule the compaction job remotely.
+                    for waiter in waiters {
+                        waiter.send(Ok(0));
+                    }
```

Comment:
When waiters are provided, we expect to send the compaction result after the compaction job is done. We should send the result after the remote job is finished.

Reply:
I remove the above `waiter.send(Ok(0))`. Instead, I add the `waiters` field in `DefaultNotifier`. Since the `waiter` can't be clonable(beacause of `Oneshot`), I have to use `std::mem::take(&mut waiters)`.

Reply:
Got it.